openmc/tallies.py

```diff
@@ -1425,25 +1426,47 @@ def export_results(self, filename='tally-results', directory='.',
             # Pickle the Tally results to a file
             pickle.dump(tally_results, open(filename, 'wb'))
 
-    def _outer_product(self, other, binary_op):
+    def hybrid_product(self, other, binary_op, filter_product='None',
+                        nuclide_product='None', score_product='None'):
         """Combines filters, scores and nuclides with another tally.
 
-        This is a helper method for the tally arithmetic methods. The filters,
-        scores and nuclides from both tallies are enumerated into all possible
-        combinations and expressed as CrossFilter, CrossScore and
-        CrossNuclide objects in the new derived tally.
+        This is a helper method for the tally arithmetic methods. It is called a
+        "hybrid product" because it performs a combination of tensor
+        (or Kronecker) and entrywise (or Hadamard) products. The filters,
+        nuclides, and scores from both tallies are combined using an entrywise
```

Comment:
I think this sentence is referring explicitly to filters, yet starts as "The filters, nuclides and scores ..."